R/scregclust.R

```diff
@@ -1367,6 +1393,63 @@ scregclust <- function(expression,
           }
         }
 
+      ###############################################
+      ## START Step 1.5: Determining target genes  ##
+      ###############################################
+
+      # With quick_mode = True a subset of target genes are selected from cycle two and onward.
+      # For each of the targets inside the noise module we determine the biggest (in absolute value)
+      # correlation to some active regulator. Then we take the top "percent" of the noise targets
+      # based on this value, together with the targets belonging to some active module, and this makes
+      # up our collection of target genes for which we calculated NNLS and do re-allocation
+
+
+      
+      if (cycle != 1 && sum(models) != 0 && quick_mode && percent != 1) {
+        # Get current active regulators and targets for subsetting the correlation matrix
+        active_regulators <- which(apply(models, 1, sum) >= 1)
+        active_targets <- which(idx != -1)
+
+        # If we are in the last cycle we only need to estimate the active targets
+        if (last_cycle){
+          filtered_targets <- active_targets
+        }
+        else{
+        # Use as.matrix to be able to use apply even when there is a single regulator
+        abs_corr_matrix_temp <- as.matrix(abs_corr_matrix[-active_targets, active_regulators])
+
+        # Create a vector of maximal absolute correlation
+        max_abs_correlation <- apply(abs_corr_matrix_temp, 1, function(row) max(row, na.rm = TRUE))
+  
+        # Find top percent
+        threshold <- quantile(max_abs_correlation, 1 - percent, na.rm = TRUE)
+  
+        # Collect index in 1:len(noise_module) which should be active
+        top_percent_targets_in_temp <- which(max_abs_correlation >= threshold)
+
+        # Obtain the index in terms all targets 1:n_target
+        top_percent_targets <- setdiff(1:nrow(abs_corr_matrix), active_targets)[top_percent_targets_in_temp]
```

Comment:
To keep it consistent with the rest of the code, replace any occurence of `1:n` with `seq_len(n)`. This works properly even if `n == 0` emitting `numeric(0)`, whereas `1:0` emits the unintuitive `c(1, 0)`.

Reply:
I see, I was not aware! Is it simply that `b:a` always goes from b to a but no check on direction is done?

Reply:
Yes, that is correct.